cli/cmd/generate_docs.go

```diff
@@ -0,0 +1,85 @@
+// Inspired by `hugo gen doc`  - see https://github.com/gohugoio/hugo/blob/release-0.69.0/commands/gendoc.go
+package cmd
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+
+type generateDocsCmdParams struct {
+	Directory  *string
+}
+
+var generateDocsParams *generateDocsCmdParams
+
+const gendocFrontmatterTemplate = `---
+date: "%s"
+title: "%s"
+slug: %s
+---
+`
+
+// crprojectCmd represents the project command
+var generateDocsCmd = &cobra.Command{
+	Use:   "docs",
+	Short: "Generates Markdown documentation for the keptn CLI.",
+	Long: `Generates Markdown documentation for the keptn CLI.
+
+This command can be used to create an up-to-date documentation of Keptn's command-line interface for https://keptn.sh.
+
+It creates one Markdown file per command, suitable for rendering in Hugo.
+`,
+	Example: `keptn generate docs
+keptn generate docs --dir=/some/directory`,
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// generate current datetime for prepender
+		now := time.Now().Format(time.RFC3339)
+
+		outputDir := "./docs"
+		if *generateDocsParams.Directory != "" {
+			outputDir = *generateDocsParams.Directory
+		}
+
+		// check if output directory exists
+		if _, err := os.Stat(outputDir); os.IsNotExist(err) {
```

Comment:
Should we mark the `dir` flag as required? Currently, when I execute this command and when I do not have a "docs" folder, I run into an error.

Reply:
Marking the `dir` flag as required does not change anything with regards to the folder not existing.

However, the behaviour is undocumented, therefore I'll make sure the default value `docs` is documented for the `dir` flag, and an appropriate error message is written when the directory does not exist.

Reply:
I've updated the documentation and added a better error message.